### PR TITLE
feat(linstor-manager): extend api to create netinterface and select preferred

### PR DESCRIPTION
**Pull request by benjamreis**
New plugin method:
- interfaceCreate(groupName, hostname, ifaceName, pifUUid): create a new net interface on a pif on a linstor node
- nodeSetPreferredInterface(gropuName, hostname, ifaceName): set a network interface as preffered for a linstor node

### CODE REVIEW

**Comment by stormi** [comment]
It would be good if the commit message would make it clear that this is about the linstor driver, and also explain in the commit message why this change is needed, what API it adds, and for what.

Also: `s/extand/extend/`

File: drivers/linstorvolumemanager.py [hunk 1]
```diff
@@ -1614,6 +1614,24 @@ def get_database_path(self):
         """
         return self._request_database_path(self._linstor)
 
+    def create_netinterface(self, hostname, iface_name, ip_addr):
```
Comment (Wescoeur) [inline drivers/linstorvolumemanager.py]: `create_node_interface` instead? It seems more uniform with the second method.

File: drivers/linstorvolumemanager.py [hunk 2]
```diff
@@ -1614,6 +1614,24 @@ def get_database_path(self):
...
+        result = self._linstor.netinterface_create(hostname, iface_name, ip_addr)
+        if not linstor.Linstor.all_api_responses_no_error(result):
+            raise LinstorVolumeManagerError(
+                'Unable to create interface on`{}`: {}'.format(hostname, ', '.join(
```
Comment (Wescoeur) [inline drivers/linstorvolumemanager.py]: Missing space: "interface on `{}`"

File: drivers/linstorvolumemanager.py (hunk 1 above)
Comment (Wescoeur) [reply drivers/linstorvolumemanager.py]: `name` instead of `iface_name`?

File: drivers/linstorvolumemanager.py [hunk 3]
```diff
@@ -1614,6 +1614,24 @@ def get_database_path(self):
...
+                    [str(x) for x in result]))
+                )
+        return json.dumps(True)
+
+    def set_node_preferred_interface(self, hostname, iface_name):
+        result = self._linstor.node_modify(hostname, property_dict={'PrefNic': iface_name})
+        if not linstor.Linstor.all_api_responses_no_error(result):
+            raise LinstorVolumeManagerError(
+                'Unable to set prefered interface on`{}`: {}'.format(hostname, ', '.join(
```
Comment (Wescoeur) [inline drivers/linstorvolumemanager.py]: `preferred`

File: drivers/linstor-manager [hunk 4]
```diff
@@ -1024,5 +1067,7 @@ if __name__ == '__main__':
         'destroyDrbdVolume': destroy_drbd_volume,
         'destroyDrbdVolumes': destroy_drbd_volumes,
         'getDrbdOpeners': get_drbd_openers,
-        'healthCheck': health_check
+        'healthCheck': health_check,
+        'interfaceCreate': interface_create,
```
Comment (Wescoeur) [inline drivers/linstor-manager]: `createNodeInterface`

File: drivers/linstor-manager [hunk 5]
```diff
@@ -1024,5 +1067,7 @@ if __name__ == '__main__':
...
+        'nodeSetPreferredInterface': set_node_preferred_interface
```
Comment (Wescoeur) [inline drivers/linstor-manager]: `setNodePreferredInterface`

**Review by Wescoeur — CHANGES_REQUESTED** [review]

**Pull request merged**